iree/compiler/Codegen/LLVMGPU/Verifiers.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/LLVMGPU/LLVMGPUUtils.h"
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "mlir/Dialect/Linalg/Passes.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+constexpr unsigned kWorkgroupTileLevel = 0;
+constexpr int kSharedMemSizeBytes = 64 * 1024;
+
+LogicalResult verifyGPUMatmulSimtPassPipeline(
+    Operation *op, IREE::Codegen::LoweringConfigAttr loweringConfig,
+    IREE::Codegen::TranslationInfoAttr translationInfo,
+    ArrayRef<int64_t> workgroupSize) {
+  auto pipeline =
+      IREE::Codegen::DispatchLoweringPassPipeline::LLVMGPUMatmulSimt;
+  StringRef pipelineName = stringifyEnum(pipeline);
+  if (workgroupSize.empty()) {
+    return op->emitOpError("expected workgroup size for GPU pipelines");
+  }
+  linalg::MatmulOp matmulOp = dyn_cast<linalg::MatmulOp>(op);
```

Comment:
any reason not to verify batch_matmul the same way it is done for the other pipeline? It is fine to do it in a separate PR but if it is the case please work on it next.